### ISSUE TITLE
Implement a cleaner way to install applications

### ISSUE THREAD

**Issue by castrojo** [ublue-os/base#5]
Once we find a declarative way to install applications we should split that off so that people deriving can have an easier time modifying what they want.

**Comment by izzthedude** [comment]
Just my two cents as a newbie:

Maybe we could have a file that lists out a bunch of flatpak app IDs where the `ublue-firstboot` program reads from during the installation process. The image maintainer (i.e you) could predefine this list with some default apps, but derived images could overwrite this file in their own Containerfiles.

Just as an example, we could create a file called `ublue-flatpaks` and `COPY` it over to `/etc/ublue/` in the Containerfile. Then during the `ublue-firstboot` process, it reads from that file and installs the apps listed in there.

If we wanna allow for more configuration for the user, we could have `ublue-firstboot` find the same file but in the home directory instead, e.g. `~/.config/ublue/ublue-flatpaks`. This way, the user could list out their own apps in here even before rebasing to the custom image.

I've actually been playing around with this idea in my fork but I've only just noticed this issue recently lmao.

**Comment by j1mc** [comment]
The issue that I have with the current method is that ... if you change your mind and want to install add'l applications (or remove applications), the 'firstboot' approach sets a file to indicate the zenity foo has already been run. With that `.config/ublue/firstboot-done` file already in place, the host OS won't pick up any changes.

I tried installing flatpak packages with Ansible during the image creation process (see my [ansible-silverblue-oci](https://github.com/j1mc/ansible-silverblue-oci) repo, but repeatedly hit issues where the build would fail if you changed anything under /var. That means no installation of flatpaks. I've tried working-around this, but haven't seen great success, and my approaches seem hack-ish - like they'll disrupt the true 'immutability' of this approach.

It's starting to seem that the current state either requires that you build your own image using the more complex ostree tooling, or only slightly modify the image using the Containerfile approach.

**Comment by castrojo** [comment]
> we could have `ublue-firstboot` find the same file but in the home directory instead, e.g.

Yeah, the container build won't let us drop stuff in the home directory, anything we do in there needs to be done after installation and the user has rebooted. 

I've been talking to @AdamIsrael about perhaps reusing this: https://github.com/Vanilla-OS/vanilla-installer

And then people can just have their own `recipe.json`. Then we can just replace the zenity thing with this and then have something that most users would expect in a first boot experience.
